src/hooks/v2v3/transactor/DistributePayouts.ts

```diff
@@ -41,11 +57,13 @@ export function useDistributePayoutsTx(): DistributePayoutsTx {
       'distributePayoutsOf',
       [
         projectId,
-        amount,
-        currency,
+        args.amount,
+        args.currency,
         ETH_TOKEN_ADDRESS, // _token
         DEFAULT_MIN_RETURNED_TOKENS, // _minReturnedTokens
-        memo,
+        versions.JBETHPaymentTerminal === ETH_PAYMENT_TERMINAL_V_3_1
```

Comment:
i feel like the neatest way to do this might be having a `useDistributePayoutsTxV3()` and `useDistributePayoutsTxV3_1()` to avoid type casting args. but tbh prolly not be worth the trouble. we'd still have a dependency somewhere on `versions.JBETHPaymentTerminal` which might make it more messy. this makes perfect sense and i'm cool with it

Reply:
Ya good call, I pretty much agree although I'm thinking it's slightly nicer to have a single `distributePayouts` hook that handles all this logic internally, that the dev doesn't have to concern themselves with "argh which version do i need". AddToBalance is a better example where there's arg factories for each version, and then depending on the version, we use the relevant args https://github.com/jbx-protocol/juice-interface/blob/74c48a6389662b09cfb29c7194c622dffab7d7cd/src/hooks/v2v3/transactor/AddToBalanceTx/AddToBalanceTx.ts#L22-L37.

Reply:
@peripheralist rolled wid it in latest commit 🙏  It's more verbose but overall clearer i think. Thanks for the call out! https://github.com/jbx-protocol/juice-interface/pull/3169/commits/3d652c35127534052845fd5b9f117a86cd181dc8#diff-5c7a868881aaa286cdc7448bd25449851402814fa24882d0f8b172fee9674c81R45